pkg/pub_integration/lib/src/fake_pub_server_process.dart

```diff
@@ -131,19 +126,25 @@ class FakePubServerProcess {
     await _stdoutListener?.cancel();
     await _stderrListener?.cancel();
     _startupTimeoutTimer?.cancel();
+    await Directory(_tmpDir).delete(recursive: true);
     if (exitCode != 0) {
       throw AssertionError('non-graceful termination, exit code: $exitCode');
     }
   }
-}
-
-/// Matches the output line.
-typedef LineMatcher = bool Function(String line);
 
-class _LinePattern {
-  final LineMatcher matcher;
-  final completer = Completer<String>();
-  _LinePattern(this.matcher);
+  Future<List<Map<String, Object?>>> readAllEmails() async {
```

Comment:
Can we leave a doc comment, saying what this returns?
```suggestion
  /// Returns a list of all emails sent by this [FakePubServer].
  ///
  /// Each email is a JSON object on the form:
  /// ```js
  /// {
  ///   "from": "<email>",
  ///   "uuid": "<message-id>", // optional
  ///   "recipients": ["<email>", ...,
  ///   "subject": "...",
  ///   "bodyText": "..."
  /// }
  /// ```
  Future<List<Map<String, Object?>>> readAllEmails() async {
```